service/src/integrationTest/kotlin/fi/espoo/evaka/application/GetApplicationSummaryIntegrationTests.kt

```diff
@@ -42,64 +56,180 @@ class GetApplicationSummaryIntegrationTests : FullApplicationTest(resetDbBeforeE
     }
 
     @Test
     fun `application summary can be be filtered by urgency`() {
-        val summary = getSummary("""{"type": "ALL", "status": "SENT", "basis": "URGENT"}""")
+        val clock =
+            MockEvakaClock(HelsinkiDateTime.of(LocalDate.of(2022, 12, 30), LocalTime.of(11, 48)))
+        val summary =
+            getApplicationSummaries(
+                user = serviceWorker,
+                clock = clock,
+                type = ApplicationTypeToggle.ALL,
+                status = setOf(ApplicationStatusOption.SENT),
+                basis = setOf(ApplicationBasis.URGENT)
+            )
         assertEquals(1, summary.total)
         assertEquals(true, summary.data[0].urgent)
     }
 
-    private fun getSummary(payload: String): Paged<ApplicationSummary> {
-        val (_, res, result) =
-            http
-                .post("/v2/applications/search")
-                .jsonBody(payload)
-                .asUser(serviceWorker)
-                .responseObject<Paged<ApplicationSummary>>(jsonMapper)
-        assertEquals(200, res.statusCode)
-        return result.get()
+    @Test
+    fun `application summary can be filtered by preschool types`() {
+        val clock =
+            MockEvakaClock(HelsinkiDateTime.of(LocalDate.of(2022, 12, 30), LocalTime.of(11, 48)))
+        fun getPreschoolApplications(vararg preschoolTypes: ApplicationPreschoolTypeToggle) =
+            getApplicationSummaries(
+                    user = serviceWorker,
+                    clock = clock,
+                    type = ApplicationTypeToggle.PRESCHOOL,
+                    status = ApplicationStatusOption.values().toSet(),
+                    preschoolType = preschoolTypes.toSet()
+                )
+                .data
+
+        assertThat(getPreschoolApplications(ApplicationPreschoolTypeToggle.PRESCHOOL_DAYCARE))
+            .extracting({ it.type }, { it.serviceNeed?.validPlacementType })
+            .containsExactlyInAnyOrder(
+                Tuple(ApplicationType.PRESCHOOL, null),
+                Tuple(ApplicationType.PRESCHOOL, PlacementType.PRESCHOOL_DAYCARE)
+            )
+        assertThat(getPreschoolApplications(ApplicationPreschoolTypeToggle.PRESCHOOL_CLUB))
+            .extracting({ it.type }, { it.serviceNeed?.validPlacementType })
+            .containsExactly(Tuple(ApplicationType.PRESCHOOL, PlacementType.PRESCHOOL_CLUB))
+        assertThat(getPreschoolApplications(*ApplicationPreschoolTypeToggle.values()))
+            .extracting({ it.type }, { it.serviceNeed?.validPlacementType })
+            .containsExactlyInAnyOrder(
+                Tuple(ApplicationType.PRESCHOOL, null),
+                Tuple(ApplicationType.PRESCHOOL, PlacementType.PRESCHOOL_DAYCARE),
+                Tuple(ApplicationType.PRESCHOOL, PlacementType.PRESCHOOL_CLUB)
+            )
     }
 
+    private fun getApplicationSummaries(
+        user: AuthenticatedUser,
+        clock: EvakaClock,
```

Comment:
Does `clock` need to be a parameter if the function is always called with the same value? I guess the same applies to the `user` parameter.

Reply:
Fixed!